app/test/vesting_test.go

```diff
@@ -0,0 +1,696 @@
+
+	assert.GreaterOrEqual(s.T(), balancesAfter.AmountOf(app.BondDenom).Int64(), minExpectedBalance)
+}
+
+// initRegularAccounts initializes regular accounts for the VestingModuleTestSuite.
+// It generates the specified number of account names and stores them in the accounts map
+// of the VestingModuleTestSuite with RegularAccountType as the key. It also generates base accounts
+// and their default balances. The generated accounts and balances are added to the provided genesis
+// state. The genesis state modifier function is returned.
+//
+// Parameters:
+// - count: The number of regular accounts to generate.
+//
+// Returns:
+// A GenesisOption function that takes the current genesis state as input, adds the generated accounts and balances to it,
+// and returns the modified genesis state.
+func (s *VestingModuleTestSuite) initRegularAccounts(count int) testnode.GenesisOption {
+	names := testfactory.GenerateAccounts(count)
+	s.accounts.Store(RegularAccountType, accountDispenser{names: names})
+	bAccounts, balances := testfactory.GenerateBaseAccounts(s.kr, names)
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range bAccounts {
+		gAccounts = append(gAccounts, &bAccounts[i])
+	}
+
+	return func(gs map[string]json.RawMessage) map[string]json.RawMessage {
+		gs, err := AddGenesisAccountsWithBalancesToGenesisState(gs, gAccounts, balances)
+		assert.NoError(s.T(), err)
+		return gs
+	}
+}
+
+// initDelayedVestingAccounts initializes delayed vesting accounts for the VestingModuleTestSuite.
+// It generates the specified number of account names and stores them in the accounts map with
+// RegularAccountType as the key. The generated accounts and balances are delayed vesting accounts
+// with varying progressive end times which enables to have a number of vesting accounts (locked) and
+// some vested accounts (unlocked balances) to test various scenarios. A modifier function which
+// modifies genesis state, including the delayed vesting accounts and their balances, is returned
+// as a GenesisOption function that takes the current genesis state as input.
+//
+// Parameters:
+// - count: The number of delayed vesting accounts to generate.
+//
+// Returns:
+// A GenesisOption function that takes the current genesis state as input, creates delayed vesting accounts
+// with varying end times using the generated accounts and balances, adds them to the genesis state,
+// and returns the modified genesis state.
+func (s *VestingModuleTestSuite) initDelayedVestingAccounts(count int) testnode.GenesisOption {
+	initCoinsForGasFee := sdk.NewCoin(app.BondDenom, sdk.NewInt(initBalanceForGasFee))
+
+	names := testfactory.GenerateAccounts(count)
+	s.accounts.Store(DelayedVestingAccountType, accountDispenser{names: names})
+	vAccounts, balances := testfactory.GenerateDelayedVestingAccounts(s.kr, names, initCoinsForGasFee)
+	gAccounts := authtypes.GenesisAccounts{}
+	for i := range vAccounts {
+		gAccounts = append(gAccounts, vAccounts[i])
+	}
```

Comment:
it does not allow for this because the types are different. The code in the loop casts a vesting account type into a genesis account type and do them in batch with `...` did not work.

Reply:
ahh I see cool, do you mind if we add something that makes that more explicit then? 
```suggestion
	for i := range vAccounts {
		gAccounts = append(gAccounts, authtypes.GenesisAccount(vAccounts[i]))
	}
```

or perhaps since we're doing this conversion multiple times we can have a simple helper function

Reply:
Good idea. will do it